### PR TITLE
PADV-1612 feat: frontend - refactor better practices

### PR DESCRIPTION
**Pull request by nandodev-net**
## Tickets

- https://agile-jira.pearson.com/browse/PADV-1612

## Description
The objective of this task is to improve the organization and maintainability of the codebase by restructuring the components directory and standardizing the order of imports according to best practices. The current setup has all components in a single directory, making it harder to manage as the project grows. Additionally, the import statements across different files need to be consistently ordered.

## Changes Made

- [x] Organize components into folders based on their usage:
    - `views/` for page-level components like `ClassRoster`, `LabSummary`, and `LabDetails`.
    - `core/` for Main component.
    - `shared/` for reusable components used across multiple views, such as `Table`, `TableFilter`, `DashboardLaunchButton`, `AlertMessage`, and `JsonViewer`.

- [x] Standardize Import Order:
    1. React and related hooks: e.g., `import React, { useState } from 'react';`
    2. Prop types: e.g., `import PropTypes from 'prop-types';`
    3. React Router: e.g., `import { useHistory } from 'react-router-dom';`
    4. Third-party libraries: e.g., `import { Breadcrumb } from '@edx/paragon';`
    5. Project utilities: e.g., `import { formatTime from '../../helpers';`
    6. Local components: e.g., `import LabDetailsCard from '../LabDetailsCard';`
    7. Styles: e.g., `import './index.scss';`

## How To Test

1. Ensure your development environment is set up with the latest dependencies by running `npm i` in your MFE folder.
2. Start the MFE by running `npm start` and ensure it is running on port 2003.
3. Access the LMS and navigate to a skillable course.
4. Click on the Training Lab tab to load the MFE.
5. **Test the Routing:**
   - Navigate through the `ClassRoster`, `LabSummary`, and `LabDetails` views using the respective UI elements.
   - Refresh the page on `LabSummary` and `LabDetails` to confirm redirection to `ClassRoster`.

Peek 2024-08-24 23-15

### CODE REVIEW

File: src/components/core/Main.jsx [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
   Redirect,
   useLocation,
 } from 'react-router-dom';
-import PropTypes from 'prop-types';
-import ClassRoster from './ClassRoster';
-import LabSummary from './LabSummary';
-import LabDetails from './LabDetails';
-import { mfeBaseUrl } from '../constants';
+
+import ClassRoster from '../views/ClassRoster';
```
Comment (01001110J) [inline src/components/core/Main.jsx:11]: Is not possible to change the way in how we import the stuff, I'm thinking in something like

`@views/ClassRoster` With this approach we do not need to use stuff like `../../` etc, here's an example to how to archive that:

https://github.com/Pearson-Advance/frontend-app-instructors-portal/blob/main/webpack.prod.config.js

https://github.com/Pearson-Advance/frontend-app-instructors-portal/blob/main/webpack.dev.config.js


Here's an example of the real usage:
https://github.com/Pearson-Advance/frontend-app-instructors-portal/blob/main/src/features/Classes/ClassDetailPage/InstructorCard.jsx#L7

Instead to do `../../../constants` we just do `features/constants` we archive that result applying the config of the files `webpack.dev.config.js` for dev mode and `webpack.prod.config.js` for production.

File: src/components/views/ClassRoster/index.jsx [hunk 2]
```diff
@@ -2,13 +2,13 @@ import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import { logError } from '@edx/frontend-platform/logging';
-import DashboardLaunchButton from '../DashboardLaunchButton';
-import TableFilter from '../TableFilter';
+import DashboardLaunchButton from '../../shared/DashboardLaunchButton';
+import TableFilter from '../../shared/TableFilter';
+import Table from '../../shared/Table';
+import { columns } from './columns';
+import { defaultErrorMessage } from '../../../constants';
```
Comment (01001110J) [inline src/components/views/ClassRoster/index.jsx:9]: I suggest to add a space here.
```suggestion

import DashboardLaunchButton from '../../shared/DashboardLaunchButton';
import TableFilter from '../../shared/TableFilter';
import Table from '../../shared/Table';
import { columns } from './columns';
import { defaultErrorMessage } from '../../../constants';
```

File: src/components/views/ClassRoster/columns.jsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Hyperlink } from '@edx/paragon';
-import { mfeBaseUrl } from '../../constants';
+import { mfeBaseUrl } from '../../../constants';
```
Comment (01001110J) [inline src/components/views/ClassRoster/columns.jsx:3]:
```suggestion

import { mfeBaseUrl } from '../../../constants';
```

File: src/components/shared/DashboardLaunchButton/index.jsx [hunk 4]
```diff
@@ -4,8 +4,8 @@ import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import { logError } from '@edx/frontend-platform/logging';
 import { Button } from 'react-paragon-topaz';
 import AlertMessage from '../AlertMessage';
-import { skillableUrl, defaultErrorMessage } from '../../constants';
-import { eventManager } from '../../helpers';
+import { skillableUrl, defaultErrorMessage } from '../../../constants';
```
Comment (01001110J) [inline src/components/shared/DashboardLaunchButton/index.jsx:7]:
```suggestion

import AlertMessage from '../AlertMessage';
import { skillableUrl, defaultErrorMessage } from '../../../constants';
```

File: src/components/views/LabDetails/index.jsx [hunk 5]
```diff
@@ -4,14 +4,14 @@ import { Redirect } from 'react-router-dom';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import { logError } from '@edx/frontend-platform/logging';
 import { Breadcrumb, Spinner } from '@edx/paragon';
-import AlertMessage from '../AlertMessage';
-import LabDetailsCard from '../LabDetailsCard';
-import LabDetailsChartCard from '../LabDetailsChartCard';
-import { formatTime } from '../../helpers';
+import AlertMessage from '../../shared/AlertMessage';
```
Comment (01001110J) [inline src/components/views/LabDetails/index.jsx:7]:
```suggestion

import AlertMessage from '../../shared/AlertMessage';
```

File: src/components/views/LabSummary/columns.jsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Hyperlink } from '@edx/paragon';
-import { mfeBaseUrl } from '../../constants';
+import { mfeBaseUrl } from '../../../constants';
```
Comment (01001110J) [inline src/components/views/LabSummary/columns.jsx:3]:
```suggestion

import { mfeBaseUrl } from '../../../constants';
```

File: src/components/views/LabSummary/index.jsx [hunk 7]
```diff
@@ -4,12 +4,12 @@ import { Redirect } from 'react-router-dom';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import { logError } from '@edx/frontend-platform/logging';
 import { Breadcrumb } from '@edx/paragon';
+import Table from '../../shared/Table';
```
Comment (01001110J) [inline src/components/views/LabSummary/index.jsx:7]:
```suggestion

import Table from '../../shared/Table';
```

File: src/components/views/LabSummary/index.scss [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /* LabSummary.scss */
 
-@import '../../variables.scss';
+@import '../../../variables.scss';
```
Comment (01001110J) [inline src/components/views/LabSummary/index.scss:3]: > [!NOTE]  
> The suggestion for aliases in `webpack` files, unfortunate does not work for `.scss`  as far I know :sob:

**Comment by 01001110J** [comment]
@nandodev-net I think I found a possible solution related with the import and the annoying linter, to fix this is required to change the folder structure related with `components` so, here's the explanation:

This is how the `webpack.dev.config.js` ( and .prod ) should look like, as you can see I removed the triple nested folder path `src/components/views` and I replaced for `src/views` :

```js
const path = require('path');
const { createConfig } = require('@edx/frontend-build');

module.exports = createConfig('webpack-dev', {
  resolve: {
    alias: {
      // you need to add core folder here too
      shared: path.resolve(__dirname, 'src/shared'),
      views: path.resolve(__dirname, 'src/views'),
      constants: path.resolve(__dirname, 'src/constants'),
      helpers: path.resolve(__dirname, 'src/helpers'),
    },
    extensions: ['.js', '.jsx'],
  },
});

```

And this is how the folder structure is organized: 

```txt
src
├── constants
│   └── content
├── core
│   ├── content
├── helpers
│   └── icontent
├── shared
    ├── content

```

I removed the folders `shared`, `core`, `views` from the folder `components` in its place I leave it into `src/` for some reason which I do not understand this webpack config is not understanding the triple nested path, maybe could be some edx config but I do not have the time to figure out, so in the package.json I installed the version `12.8.57` of the package `@edx/frontend-build` doing this will throw a new issues related of the organization of the code, for instance, you will have this error in `Main.jsx`:

> [!CAUTION]
> React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be 
> statically checked.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)


Once you complete this config you shouldn't see any other path error, ( maybe new warnings related code organization ), so hope this help.

**Review by JuanDavidBuitrago — APPROVED** [review]
Its working well

**Review by 01001110J — APPROVED** [review]

**Pull request merged**